CMakeLists.txt

```diff
@@ -60,7 +60,18 @@ option(BUILD_BENCHMARKING "Build all benchmarks." OFF)
 option(BUILD_PYTHON_BINDINGS "Build Python bindings with pybind11." OFF)
 option(BUILD_TESTING_PYTHON "Build Python tests only." OFF)
 
+# Include the vendored FindPython3 resources
+list(PREPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+# Find a suitable Python installation
+find_package(Python3 COMPONENTS Interpreter Development.Module)
+
 if (BUILD_PYTHON_BINDINGS)
+
+  if (NOT Python3_FOUND)
+    message(FATAL_ERROR "Failed to find a compatible Python3 installation")
+  endif()
```

Comment:
> Could this be rearranged so that,
> 
>     * we try to find Python3 only if necessary

I proceeded with the current implementation because Python is needed 1) when the bindings are created and 2) when the Python tests are enabled. The find logic can be guarded with the flags used to include these two subfolders.
 
>     * we first try to find Python3 with the system's CMake modules and only use the vendored one if it fails - similarly to Eigen [above](https://github.com/artivis/manif/blob/cd1d1403430ce56ae1d371eb82f3e287c9a4e82f/CMakeLists.txt#L28)??

I guess something similar could be done, even if I don't see the point of not using the vendored files all the times since they're there. The only problem I see in using system's modules first is that often a Python interpreter is detected among the many options in the system, but is the wrong one, or one that does not have all the components. Therefore, I see the fallback method as not very robust. The worse case is when the interpreter and the libs are detected with different versions, and this could happen in very old CMake versions. Just to understand your point, what kind of problem do you expect when only the vendored files are used? It seems to me a good practice to provide a more uniform behavior without complicating the logic.

Reply:
Could 2620291 be ok?

Reply:
Concerning the first point, looks good :+1:.
As for the second, I usually prefer to rely on what's shipped on the system by default, that's how thing should be and we shouldn't need to vendor. However in Eigen case it is simpler, it's binary, `find_package` works or not. In Python3 case, if there is also the possibility of a 'messed up' detection then I guess it make sense to rely solely on the vendored cmake files.